src/Core/Checkout/Document/Renderer/InvoiceRenderer.php

```diff
@@ -163,4 +169,28 @@ private function getNumber(Context $context, OrderEntity $order, DocumentGenerat
             $operation->isPreview()
         );
     }
+
+    private function isAllowIntraCommunityDelivery(array $config, OrderDeliveryEntity|null $orderDelivery = null): bool
+    {
+        if (empty($config['displayAdditionalNoteDelivery']) || empty($config['deliveryCountries'])) {
+            return false;
+        }
+
+        if (!$orderDelivery) {
+            return false;
+        }
+
+        /** @var OrderAddressEntity $shippingAddress */
+        $shippingAddress = $orderDelivery->getShippingOrderAddress();
+
+        $country = $shippingAddress->getCountry();
+
+        if (!$country) {
+            return false;
+        }
+
+        $isCompanyTaxFree = $country->getCompanyTax()->getEnabled();
```

Comment:
Why do you need to check the tax state here? 

In the settings for the invoices there is a config option to configure for which countries this note should be added, IMHO we should only check that configuration

![image](https://github.com/shopware/shopware/assets/15930605/a9471db2-6a28-4b3d-a5bb-a071b5dd63f2)

Reply:
<img width="753" alt="Bildschirmfoto 2024-01-09 um 10 03 48" src="https://github.com/shopware/shopware/assets/60429589/2b0fb785-2013-4476-a5bf-d6f3852a4678">

(only found the german documentation is clearly stating what should be done 😄)

Your documentation says that the "Tax-free (B2B) property should be enabled, so that this label is displayed on the document. Therefore this check is necessary in my imo.